node/finder.ts

```diff
@@ -4,13 +4,20 @@ import { basename, extname, join } from 'path'
 import { EdgeFunction } from './edge_function.js'
 import { nonNullable } from './utils/non_nullable.js'
 
-const ALLOWED_EXTENSIONS = new Set(['.js', '.ts'])
+const ALLOWED_EXTENSIONS = new Set(['.js', '.jsx', '.ts', '.tsx'])
 
 const findFunctionInDirectory = async (directory: string): Promise<EdgeFunction | undefined> => {
   const name = basename(directory)
-  const candidatePaths = [`${name}.js`, `index.js`, `${name}.ts`, `index.ts`].map((filename) =>
-    join(directory, filename),
-  )
+  const candidatePaths = [
+    `${name}.js`,
+    `index.js`,
+    `${name}.ts`,
+    `index.ts`,
+    `${name}.jsx`,
+    `index.jsx`,
+    `${name}.tsx`,
+    `index.tsx`,
+  ].map((filename) => join(directory, filename))
```

Comment:
How do you feel about using `ALLOWED_EXTENSIONS` to generate all candidate paths, as opposed to enumerating them?

```suggestion
  const candidatePaths = [...ALLOWED_EXTENSIONS]
    .flatMap((extension) => [name + extension, `index${extension}`])
    .map((filename) => join(directory, filename))
```

Reply:
Yeah, that is better